mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMessageMapper.java

```diff
@@ -191,12 +136,22 @@ public long countUnseenMessagesInMailbox(Mailbox mailbox) throws MailboxExceptio
      */
     public void delete(Mailbox mailbox, MailboxMessage message) throws MailboxException {
         try {
-            getEntityManager().remove(message);
+            AbstractJPAMailboxMessage jpaMessage = getEntityManager().find(AbstractJPAMailboxMessage.class, buildKey(mailbox, message));
+            getEntityManager().remove(jpaMessage);
+
         } catch (PersistenceException e) {
             throw new MailboxException("Delete of message " + message + " failed in mailbox " + mailbox, e);
         }
     }
 
+    private AbstractJPAMailboxMessage.MailboxIdUidKey buildKey(Mailbox mailbox, MailboxMessage message) {
```

Comment:
The AbstractJpaMailboxMessage is the base Message model which contain all needed properties but base on advanced feature (Streaming or Encryption) we can support message which has header and body with stream or encrypted.
But for now, with guice, we just support default message. So do we need to keep them all as they are or remove all of them, remove abstract, just keep default model?